src/components/dagshub/data-engine/queryBuilder/NewDatasetNameModal.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react';
```

Comment:
I don't see a reason this should be in the design system, doesn't really seem reusable for anything else.